### ISSUE TITLE
Memory Leak on Latest Dev Builds

### ISSUE THREAD

**Issue by RedZephon** [dmulloy2/ProtocolLib#1272]
I'm getting a memory leak that crashes the server on boot using the latest dev builds. I tried a few different dev builds and it seems to persist on all the Java 16 builds.

I went through and removed all plugins and added them back 1 by 1 to narrow it down and can confirm that ProtocolLib is the issue. Removing it allows the server to boot. Here's a log of a failed boot, it appears to be a memory leak.

[2021-06-26-9.log](https://github.com/dmulloy2/ProtocolLib/files/6720882/2021-06-26-9.log)

I tested everything on a Dev Server and it seemed to work, these errors only started happening when I moved it to Production which has different hardware.

**Comment by dmulloy2** [comment]
I don't think a memory leak would appear that quickly. Generally those show up after a few hours of running. How much RAM are you running your server with? Could you possibly run a profiler and get a heap dump?

**Comment by RedZephon** [comment]
My server has 10gb of RAM. I've been working with the itemsadder dev since the issues also seem to be linked to his plugin (disabling itemsadder stops crashes from happening). He says its not his problem and cant reproduce.

I was able to get things working by deleting the itemsadder folder and have it regenerate and that seemed to work but after some time I got the crashes again. I was able to run a heap dump but it hasnt provided anything useful for me. Here is the dump though. https://www.dropbox.com/s/3735zjkpvnp12pb/jul1.hprof?dl=0
